src/schema/auth.ts

```diff
@@ -0,0 +1,130 @@
+import { z } from "zod";
+import { v4 as uuidv4 } from "uuid";
+import { AddressSchema, RawDateSchema } from "./shared";
+
+/**
+ * @internal
+ */
+export const AuthenticationPayloadInputSchema = z.object({
+  /**
+   * The application to authenticate to
+   */
+  application: z.string(),
+  /**
+   * The address of the wallet requesting to authenticate
+   */
+  subject: AddressSchema,
+  /**
+   * The audience of endpoints intended to receive the authentication payload
+   */
+  audience: z.array(z.string()).optional(),
```

Comment:
should we just call this endpoints?

Reply:
Hmm - I do think this field could use a rename potentially (although I think deferring to the JWT is a good default if we have nothing else) - but I think audience may be too limiting. The intention of the `aud` field on JWT, although it's often used to specify which endpoints a token can auth to, isn't always used like that.

In fact in the JWT spec, audience is supposed to be a field that's open to whatever use the backend wants to use it for.

Reply:
Ill change it to `endpoints` for now for simplicity - if we need to expand scope later we can